### ISSUE TITLE
"Reverse time graphics" option removed

### ISSUE THREAD

**Issue by jostephd** [wesnoth/wesnoth#1905]
Hi,

1.13.8 no longer supports the "Reverse time graphics" option in Preferences -> Advanced (`flip_time=yes` in .config/wesnoth/preferences).  1.12.6 did support it.

Could that be re-added please?  I'm used to it, and besides, I'm in the northern hemisphere (north to the Tropic of Cancer) and my window is facing south, so reverse time graphics makes the sun in the game go in the same direction (left to right) as the sun in the window. :-)

**Comment by Wedge009** [comment]
It was indeed in 1.12 and I confirm it's missing (even with manual editing of the preferences file to add flip_time) in master. I don't know if the removal of this feature was deliberate or not, though, but for the time being I would call this a regression.

**Comment by gfgtdf** [comment]
Iirc @Vultraz expicitly removed this feature, but you are not the first one who complained about it missing, so it's probably better to readd it.

**Comment by Vultraz** [comment]
Yes, I deliberately removed it. Even if the graphics weren't going to be redone (which they are), I still would not support re-adding this. I'm sorry.

**Comment by jostephd** [comment]
@Vultraz Thanks for the heads up about the new graphics.  Could you explain the reason why the option shouldn't be re-added?  If we understood your concerns, we might be able to come up with an alternative that addresses those shortcomings.  Thanks again.

**Comment by jostephd** [comment]
I'm currently using the following instead of the flip_time option:

```
# Reverse the graphics of the default time-of-day schedule
$ for i in /usr/share/games/wesnoth/1.13/data/core/images/misc/time-schedules/default/*; do
  sudo mogrify -flop $i
done
```

**Comment by Wedge009** [comment]
I would also be interested in knowing the reason for removing this, even though I don't use it myself. Does it cause problems with something else? Just wondering.

**Comment by Vultraz** [comment]
Honestly, I'm being rather petty here. It just seemed like such a ridiculous option to me so I dropped it a time when I was going through and trying to remove unnecessary preferences.

**Comment by jostephd** [comment]
Thanks for the explanation.  I'd be the first to support removing unused code.  On the other hand, the codepath does have users (me and the other people @gfgtdf mentioned), and doesn't seem to be adding a large maintenance burden, so perhaps reviving it could be justified.

I had a look at what that would involve.  The commit that removed flip_time is db8c65aad8ef998ca0c02e92432a086f77b187aa; reverting that commit restores the functionality.

To cut a long story short, I'd appreciate it if the removal were reconsidered.

**Comment by jostephd** [comment]
Reverting db8c65a had a few conflicts; I resolved them here: https://github.com/wesnoth/wesnoth/compare/master...jostephd:restore-reverse-time-graphics?expand=1

**Comment by Wedge009** [comment]
I think it is reasonable to have this option. I see it as similar to how different hemispheres view the phases of the moon. I don't think it's frivolous or superfluous at all. If at least one user wants it and it's not terribly hard to maintain and it doesn't conflict with any technical frameworks, I think it's worth keeping.

**Comment by Vultraz** [comment]
I'm sorry, but I'm vetoing re-adding this. Yes, it's a minor thing, but it's really an case of Options Are Bad. It doesn't add anything to the game experience and basically exists for the sake of existing. I mean, hell, in one of the game's epochs there are *two suns*, so believing the planet rotates a certain way isn't even that fantastical.

**Comment by jostephd** [comment]
> It doesn't add anything to the game experience

It does for me.  I wasn't kidding in my first post when I said my window faces south and the sun (the real one) moves left to right as I'm playing, so the UI is literally confusing when this option is unset.  The option changes my UX for the better.

I am not asking to make the game either more fantastical or less fantastical.  I am asking to add an option that makes the UI communicate the time of day differently.

Thanks for the discussion.  I am trying to convince you to change your mind, but I don't want to come across as beating a dead horse.

**Comment by jostephd** [comment]
Alternatively, how about dropping the option but mirroring the images for everyone, so the sun in the game would go left-to-right for everyone?

**Comment by Vultraz** [comment]
That might be acceptable.

**Comment by Wedge009** [comment]
So... no love for southern hemisphere?

**Comment by jostephd** [comment]
@Wedge009 I sympathize with users in the southern hemisphere; I _have_ walked a mile in their shoes.  If the default sun direction is changed to left-to-right, they will be able to use the workaround and patch I posted above to change it back.

I think that defaulting to left-to-right is better than to right-to-left because there are more users in the northern hemisphere than in the southern one.  I think reviving the flip_time option would be better than hardcoding either sun direction, and I think you agree, but...

**Comment by CelticMinstrel** [comment]
Does this even have anything to do with hemispheres? It's not like the sun moves differently in the southern hemisphere!

**Comment by gfgtdf** [comment]
in the sourhers hemispsere the sun is midday in the north so the pichte is drawn looking towards north while the nothern picture is drawn looking towards south.

(in fact this isn't 100% correct, where the sun is at midday not only depends on the hemispere but also on the season)

**Comment by mattsc** [comment]
The sun doesn't move differently, but the viewpoint it different.  And yes, that does make it look like it is going through the sky in the opposite direction (as long as you're farther south than where it crosses the zenith at the time of year, as gfgtdf says).

Life would be much easier if we actually lived on Discworld!

**Comment by jostephd** [comment]
When you're **north** to the Tropic of Cancer, the sun is in the south half of the sky, and appears to move from **left (east)** to **right (west)**.

When you're **south** to the Tropic of Capricorn, the sun is in the north half of the sky, and appears to move from **right (east)** to **left (west)**.

**Comment by CelticMinstrel** [comment]
Okay, I suppose that makes sense. That said, I don't see why we need to associate the in-game sun with the real sun.

**Comment by pydsigner** [comment]
@CelticMinstrel makes it easier to read the ToD graphic. It has definitely thrown me off before. If you have to mouse over the ToD to remember whether it's morning or afternoon, there's not all that much point to the ToD graphic whatsoever.

**Comment by CelticMinstrel** [comment]
Ah, I suppose you do kinda have a point there...

**Comment by jgomo3** [comment]
The discussion should not be about the Real World Sun, but about the scenery world.

The feature is a minor detail anyway. It could be a nice to have detail so, in a campaign where things happens in the north hemisphere, the animation goes in one direction but in scenarios happening in the south hemisphere, the animation goes in the another.

And in worlds with 2 or more suns, or many moons, the animation could be adjustable. But in any case, I think this is just a "almost nice to have" feature.

**Comment by vgaming** [comment]
I miss the old functionality a lot, too. (I only saw this issue now, but I did notice the change itself instantly.) I think this issue is especially painful for people that _know_ how to orient/navigate by sun, but their habits get suddenly broken.

The personal impact for me is that I started to always _hover_ the ToD image. This seems like the worst possible outcome. So there is a really huge icon, but it's not used. There is no label also, so you have to make mouse movements.

Honestly, I really, really don't see a maintenance burden here.

**Comment by tuxlifan** [comment]
**TL;DR**:
Knowing the time of day is in my view *crucial*. All players are biased by either their real world experience or reading direction preference.
Apart from it being "another option" (couldn't we put it in Advanced then? And its quite a senior one, isn't it? ;)) are there any technical burdens? Maybe other ways to still give the user the choice, e.g. during installation?

I am obviously quite late to the party :S but this touches for me one of the *core* aspects of playing Wesnoth with a strategy in mind, so here you have my 2¢:

@jgomo3 I heartfully disagree regarding the importance. The knowledge of the time of day has in my (past) experience been crucial to avoid catastrophic losses.
Any player *will* have a bias either from their knowledge/observation of the real world sun and its apparent movement or from their usual reading direction of their language.

I find it not unusual to check the time of day multiple times during one turn alone when switching between various classes/units and it is most often very important for me to know if a unit can take 1 or 2 turns to reach a safe terrain, etc.

Before the toggle was implemented back in the days I would routinely mess things up because thinking about translating the direction of the sun/moon was just too much mental work and my units (mostly Elves :)) would get surprised by a sudden unexpected nightfall. No fun when there are Orcish Riders out to get you...
Whenever I installed Wesnoth on a new computer or started a new "gaming phase" with it (without a previous profile saved), the first thing I *always* did was go to fix the direction of the time graphics for my view of the sun.

Now, to try to contribute also rather than only lament (thanks for reading this far!):

I don't understand fully what the issue is for the maintenance burden nor what the ~FL() code does, presumably it flips the graphic while displaying.
If the problem is that it's cumbersome to do the flipping with the new 2.0 SDL how about adding already flipped graphics and then loading either time_of_day_SH_<id> or time_of_day_NH_<id> images (for Southern/Northern Hemisphere)?
Or how about a command line option? Or editing the settings file by hand?
Or making an option/script that allows to switch it during install, if you really want to avoid having a preference item for it at all (although I strongly suspect that a preference item might keep the need for explanation lower)?

To be blunt: I personally can't enjoy Wesnoth without proper time graphics so I will, if things stay the same, have to do the workaround by @jostephd (thanks!) and I respect that I can't demand anything of you but I would really love to find a solution where my gaming friends can finally just download Wesnoth from Steam and play with me in an easy manner (previously installing and all having the same version was kind of "too much of a hassle" for them, so thanks for making that part easier already!).

I agree that in-game world dynamics/physics could and should modify how or what graphics would be displayed but at the same time the player *just needs* a solid way to tell the time of day reliably and if that would mean to scratch those nice graphics and find a different way how to depict it, IMHumbleO.

Thanks!

**Comment by Ja-MiT** [comment]
It seems to me that this should be viewed more along the lines of localization (making the interface readily understandable) than along the lines of a gameplay option. The interface uses these images to convey information to the player. So the game becomes more accessible to more people if the image can be "localized", removing the need for players to "translate" the image to their accustomed understanding.

(While this sort of localization is a bit out-of-the-box, it is not unheard of to localize images to better convey information to people of different cultures. However, since cultures and hemispheres do not necessarily correlate well, an option was needed for the sun's direction.)

As for arguments about being accurate to the fantasy world, I would say they are misguided. The primary purpose of these images is to convey information, not to provide an accurate view of the Wesnothian sky. Incorporating elements such as a second sun add flavor, but this should not go so far as to obscure the information the image is supposed to convey.

**Comment by Landiss** [comment]
@Vultraz your change only affects night, but the issue is with sun (for me at least, but that's how I understand other comments here too).

I honestly have trouble understanding what's the reasoning of Wesnoth sun going the other way around than in our world, other than someone doing this incorrectly in the first place (simple mistake, because east is on the right side of the map etc.) and everyone getting used to it. I imagine the only reason why the fix to this was introduced as an option (and not simply a change) must have been the fact that people got used to it.

With the additional counter next to the ToD graphics it is slightly better nowadays, but the graphic itself is still misleading for me.

@Vultraz I understand you want to limit the length of the advanced options table in game. There are options in there that I imagine are not of interest to a regular player (e.g. logging, graphic scaling, damage calculation type). What if they were removed from GUI and made accessible only via settings file? You could restore the ToD reverse option in the same way then.

**Comment by ProditorMagnus** [comment]
I agree, swap all or nothing, but dont leave this inbetween state where night and day use different way.

**Comment by jostephd** [comment]
@Vultraz Thanks for swapping the first and second watch graphics. However, now the moon is [on the left in schedule-dusk.png](https://github.com/wesnoth/wesnoth/blob/master/data/core/images/misc/time-schedules/default/schedule-dusk.png), but [on the right in schedule-firstwatch.png](https://github.com/wesnoth/wesnoth/blob/master/data/core/images/misc/time-schedules/default/schedule-firstwatch.png). Moreover, the moon in the default schedule now moves in a different direction to the sun in the default schedule (and to the sun _and_ moon in [tod-schedule-24hrs.png](https://github.com/wesnoth/wesnoth/blob/master/data/core/images/misc/time-schedules/tod-schedule-24hrs.png)).

I agree with @ProditorMagnus that the sun and moon should move in the same direction.

**Comment by Vultraz** [comment]
Oh, hell. I guess I'll just revert my commit, then. I'm not going and reconstructing new images.

**Comment by ProditorMagnus** [comment]
Leave the issue open in case someone is willing to create the images.

**Comment by Vultraz** [comment]
No.

**Comment by Vultraz** [comment]
And to clarify, that is because if we're going to work with anyone on new ToD images, it should be working with LordBob. The "new" default ones I constructed from images he did, and yes, I know the 24 hour schedule now looks completely out-of-place. But I don't have the image templates, nor do I have the ability to draw anywhere near the quality LordBob can.

And FTR, the image revamp I mentioned last year still hasn't technically happened yet; my updates to the default schedule were an improvement, but a new set would be at a larger size, at the very least (and include the 24h schedule) to fit in with my planned HUD revamp in 1.15.

I'll keep this issue in mind if I ever fix the issue on master blocking my progress on the HUD redesign. If we use the images from which I extracted, the updates to the default schedule pre-1.14, the celestial progression will likely stay the same, unless LB happens to still have the templates after all these years. If he wants to redo them, I can ask him to alter the direction. But it's definitely not something we want a random contributor doing.

And yes, I've locked this conversation because I really would rather not argue about it further. It's unfortunate my swapping of first and second watch didn't suffice. I had hoped it would. Oh well. :(

**Comment by jyrkive** [comment]
Everyone: Ravana created an addon "Mirror Schedule Images" that, when installed, loads mirrored time of day images instead.

**Comment by sevu** [comment]
When this happens, maybe one could add two new images which are for a hex which is for whatever unnatural reason enlightened or darkened permanently. It would need to be a _single_ image which fits to every of the other ToD images.
A place where it could be used would be the central hex in the MP map Tombs of Kesorak.

**Comment by CelticMinstrel** [comment]
How is that different from the image that's automatically overlaid when you mouse over a hex affected by illuminates?

**Comment by sevu** [comment]
The overlay signalizes a ToD change by +25%. In this case you can have "night" at some hexes while it is day at another – the overlay is not enough to display the difference of 50 % points.

In addition, there is the case that a Mage of Light could stand next to it, where the overlay should be applied over said image, with the resulting ToD modifier being 0%.

The situation is the same like cave terrain next to normal map, with the difference that the cave ToD image doesn't fit, as it is under the open sky.

[same applies for the darkening instead of lightening]
